web/webserver.go

```diff
@@ -50,3 +52,17 @@ func (s *QuickFeedService) RegisterRouter(tm *auth.TokenManager, authConfig *oau
 
 	return router
 }
+
+// controller is a wrapper for the QuickFeedService handler that sets a write deadline for the submission stream.
+// TODO: Remove this when connect-go finally supports deadlines.
+// TODO: https://github.com/connectrpc/connect-go/issues/604
+func controller(h http.Handler) http.Handler {
+	timeout := 15 * time.Minute
```

Comment:
We should not hard code the timeout herein. It becomes slow to test. If you pass the timeout as an argument to `contoller` it can more easily be tested.

Moreover, it would be nice to have a test for this, so that Codecov can be made happy as well.